src/puppetlabs/puppetdb/query_eng.clj

```diff
@@ -168,44 +169,46 @@
                                (update :order_by paging/munge-query-ordering))
         query-options (merge (dissoc query-map :query) paging-options)]
 
-    (try
-      (jdbc/with-transacted-connection scf-read-db
-        (let [munge-fn (get-munge-fn entity version query-options url-prefix)
-              {:keys [results-query count-query]} (-> remaining-query
-                                                      coerce-from-json
-                                                      (query->sql entity version query-options))
-              query-error (promise)
-              resp (http/streamed-response
-                    buffer
-                    (try (jdbc/with-transacted-connection scf-read-db
-                           (jdbc/with-query-results-cursor
-                             results-query (comp #(http/stream-json % buffer pretty-print)
-                                                 #(do (when-not (instance? PGobject %)
-                                                        (first %)) (deliver query-error nil) %)
-                                                 munge-fn)))
-                         (catch java.sql.SQLException e
-                           (deliver query-error e))))]
-          (if @query-error
-            (throw @query-error)
-            (cond-> (http/json-response* resp)
-                    count-query (http/add-headers {:count (jdbc/get-result-count count-query)})))))
-      (catch com.fasterxml.jackson.core.JsonParseException e
-        (log/error
-         e
-         (i18n/trs "Error executing query ''{0}'' with query options ''{1}''. Returning a 400 error code."
-                   query query-options))
-        (http/error-response e))
-      (catch IllegalArgumentException e
-        (log/error
-         e
-         (i18n/trs "Error executing query ''{0}'' with query options ''{1}''. Returning a 400 error code."
-                   query query-options))
-        (http/error-response e))
-      (catch org.postgresql.util.PSQLException e
-        (if (= (.getSQLState e) "2201B")
-          (do (log/debug e (i18n/trs "Caught PSQL processing exception"))
-              (http/error-response (.getMessage e)))
-          (throw e))))))
+    (if (:parse_pql_only query-options)
+      (http/json-response query)
```

Comment:
@ajroetker And as for moving it to the top level of the function, do you mean before the 'let'? because I had thought about that and I think I talked to @wkalt about it, but I think we concluded that it is fine after?